### ISSUE TITLE
Testing Float datatypes

### ISSUE THREAD

**Issue by pattonw** [pattonw/rust-pyn5#10]
### Improve testing of float datatypes

Problems with datasets of type float
1) No error is thrown when writing an array of ints to a dataset of type float, I assume python casts ints to floats somewhere.
2) When casting to floats, small changes may occur such as 0.1 to 0.100000001490116119384765625.
3) Overflow does not happen as expected because large floats are just written as inf

**Comment by clbarnes** [comment]
> When casting to floats, small changes may occur such as 0.1 to 0.100000001490116119384765625.

Could this just be floats being floats? It seems like it may be more of a "how different languages format numbers by default" issue than anything else.

**Comment by pattonw** [comment]
Yeah, it is just floats being floats. But I just wasn't sure if n5 is used in cases where floating point precision might be an issue. You should always be able to write and read float64 exactly, but should we throw a warning or error if you try to write ints to a float dataset which works, but is imprecise

**Comment by clbarnes** [comment]
That's a good call. Given numpy arrays have strict types which can easily be cast between anyway, I'd throw an error if someone tried to write an int array to a float dataset - make the caller do `my_int_array.astype(float)` if that's what they want (explicit is better than implicit).

**Comment by pattonw** [comment]
Thats what I was thinking. Its not a huge priority for now, so I'll leave it with expected fail tests. I think this problem should resolve itself when we build in numpy support to the rust side
